2016-02-05  Yyxxxx Ch  <classxxx@lnxsrv09.seas.ucla.edu>

        * timespec.h (timespec_cmp): Fix the bug that ls -t mishandles files whose time stamps are very far in the past.

--- /u/eng/class/classxxx/35L/week5/lab5/newTar/coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2016-02-05 15:53:06.466174000 -0800
@@ -45,8 +45,19 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  else
+    {
+      if (a.tv_nsec < b.tv_nsec)
+	return -1;
+      else if (a.tv_nsec > b.tv_nsec)
+	return 1;
+      else
+	return 0;
+    }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
